core/src/main/java/org/infinispan/remoting/responses/ClusteredGetResponseValidityFilter.java

```diff
@@ -33,8 +33,8 @@ public ClusteredGetResponseValidityFilter(Collection<Address> targets, Address s
    public boolean isAcceptable(Response response, Address address) {
       if (targets.contains(address)) {
          missingResponses--;
-         if (response instanceof SuccessfulResponse) {
-            validResponses++;
+         if (response instanceof SuccessfulResponse || response instanceof ExceptionResponse) {
```

Comment:
@tristantarrant Turns out our current behaviour is actually 2), with the caveat that we delay throwing the exception until we get a response from all the owners. We started behaving like 1) after the staggered gets work, but because of a bug that happened only when staggered gets were actually disabled. 

Turns out supporting 1) and reporting meaningful exceptions is also pretty awkward with the `ResponseFilter` API, so I agreed with @rvansa to converge to 2) everywhere, at least until we implement our own response handling on top of `JChannel`.